flink-core/src/main/java/org/apache/flink/configuration/CheckpointingOptions.java

```diff
@@ -114,6 +114,25 @@
                                     + "if no value set (job or cluster), the change log will not be "
                                     + "enabled.");
 
+    /** Which storage to use to store state changelog. */
+    @Documentation.Section(value = Documentation.Sections.COMMON_STATE_BACKENDS)
+    public static final ConfigOption<String> STATE_CHANGE_LOG_STORAGE =
+            ConfigOptions.key("state.backend.changelog.storage")
+                    .stringType()
+                    .defaultValue("memory")
```

Comment:
We can change this to `InMemoryStateChangelogStorageFactory.getIdentifier()`

Reply:
This also makes ```flink_core``` depend on the ```flink_runtime```.
I leave it and add some javadoc.

WDYT?

Reply:
I forget the dependency relationship, and I think it would be okay here to not let `flink-core` depends on `flink-runtime`.